### ISSUE TITLE
Second level multifieldpanel

### ISSUE THREAD

**Issue by GR1ML0CK** [Adobe-Consulting-Services/acs-aem-commons#380]
when using the multifieldpanel xtype inside of another multifieldpanel, the second level one adds extra qoutes and escape characters (ie: backslashes). This is causing an issue with using widgets:getMultiFieldPanelValues. it does not map this second level and im not able to access those keys.

example of what gets generated:
name: tabEntries
value:

``` json
{"tabName":"xxx","tabIcon":"aaa","accordion":["{\"itemName\":\"eec\",\"itemIcon\":\"fff\"}","{\"itemName\":\"ddd\",\"itemIcon\":\"iii\"}"]}
```

This works: 

``` jsp
<c:set var="tabEntries" value="${widgets:getMultiFieldPanelValues(resource, 'tabEntries')}"/>
<c:forEach items="${tabEntries}" var="tabEntry">
                ${tabEntry.tabIcon}
                    ${tabEntry.tabName}
</c:forEach>
```

but this does not:

``` jsp
<c:set var="tabEntries" value="${widgets:getMultiFieldPanelValues(resource, 'tabEntries')}"/>
<c:forEach items="${tabEntries}" var="tabEntry">
                <c:forEach items="${tabEntry.accordion}" var="item">
                      ${item.itemName}
                </c:forEach>
</c:forEach>
```

**Comment by cameronoxley** [comment]
+1

**Comment by msawenkosibrian** [comment]
+1

**Comment by justinedelson** [comment]
What do you mean "it does not work"? What does that second example output?

**Comment by paul-bjorkstrand** [comment]
tl;dr: inner multifields stored as string value in outer multifield in classic, but stored as json objects in touch.

This is due to a difference in how the content is serialized inside a multifield, if I am reading the code correctly. From what I remember about the classic multifieldpanel xtype,  inner multifields are serialized first as a string value, which is placed in the form. Then outer multifields place this string value into their own structure. When the resulting object is serialized to a string, the inner multifield has been serialized _twice_.

From what I can tell from the touch version, nested multifields are handled by collecting all the data in a single object, with nested multifields' _objects_ (unserialized) being stored into the outer multifield then being serialized. This means the inner multifield is serialized only once.

Example:

Given the pseudo form, with values below
```html
<form>
	<div data-name="outer_multifield">
		<div class="item">
			<input name="field_1" value="outer_1_field_1"/>
			<div data-name="inner_multiield">
				<div class="item">
					<input name="field_1" value="inner_1_field_1"/>
					<input name="field_2" value="inner_1_field_2"/>
				</div>
				<div class="item">
					<input name="field_1" value="inner_2_field_1"/>
					<input name="field_2" value="inner_2_field_2"/>
				</div>
			</div>
		</div>
	</div>
</form>
```

Classic will produce the following json when serialized:
```json
{
  "field_1": "outer_1_field_1",
  "inner_multifield": [
    "{\"field_1\":\"inner_1_field_1\",\"field_2\":\"inner_1_field_2\"}",
    "{\"field_1\":\"inner_2_field_1\",\"field_2\":\"inner_2_field_2\"}"
  ]
}
```
Touch will produce the following json when serialized:
```json
{
  "field_1": "outer_1_field_1",
  "inner_multifield": [
    {
      "field_1":"inner_1_field_1",
      "field_2":"inner_1_field_2"
    },{
      "field_1":"inner_2_field_1",
      "field_2":"inner_2_field_2"
    }
  ]
}
```
The last time I used the classic version, the nested multifield content actually broke some of AEM's built in features (move page process, for example), which led us to reworking any component that used nested multifields, to eliminate them. If what I know and have read about the touch version is accurate, the differences in JSON structure would be significant enough to cause major problems when moving from classic to touch

**Comment by davidjgonzalez** [comment]
in support of a deprecated ClassicUI .. closing as wont fix

**Comment by paul-bjorkstrand** [comment]
While I understand the desire to fully deprecate classic, this will turn into a huge barrier to migrating from classic to touch. I feel that closing for that reason is rather short sighted.

**Comment by badvision** [comment]
Feel free to re-open with a PR to solve if you'd like.  It's just not in our best interests going forward to encourage folks to keep using classic UI.  It is going away.  People need to stop using it.

**Comment by davidjgonzalez** [comment]
@paul-bjorkstrand the biggest reason this is closed is because no one felt it was important enough to fix (provide a PR) since it was opened in 2014. By all means, PR away for anything you please :)

We really need to clean-up our issues as theyre getting crusty to the point its hard to figure out what relevant things to work on are. If you feel super-strongly thought, re-open and ideally provide a PR to fix.
